src/i18n/format.ts

```diff
@@ -10,9 +10,16 @@ export default function(value: any, format?: string, lng?: string) {
         locale: getDateLocale(lng)
       });
     } catch (e) {
-      return dateFormat(isoDate as Date, DEFAULT_FORMAT, {
-        locale: getDateLocale(lng)
-      });
+      if (e instanceof RangeError) {
+        // eslint-disable-next-line no-console
+        console.error(e.message);
+        // given date format was invalid, format with the default one
+        return dateFormat(isoDate as Date, DEFAULT_FORMAT, {
+          locale: getDateLocale(lng)
+        });
+      } else {
+        throw e;
```

Comment:
I'm already trying to propose a middle-ground here, I'd personally tend to throw in any case; period.
I'd love hear @JalilArfaoui opinion on this.

Reply:
My general opinion is that we should throw on anything catchable at built time. Anything we can’t check at build time, we should handle at runtime so that we have a fallback.

I’m OK with this version, since an incorrect `format` given to `dateFormat` will throw a `RangeError` (https://date-fns.org/v2.22.1/docs/format#exceptions)

So we should still avoid breaking all the app in case of bad `format` 

I’m not even sure which case could end up in the `else { throw e }` ? 

Reply:
> So we should still avoid breaking all the app in case of bad format

Just to be clear, we are supposed to detect bad format at build time since this function is used in "static" translation.
We might miss a few tests to detect it early on tho...